aiinfra-cluster/modules/gke-cluster/main.tf

```diff
@@ -0,0 +1,284 @@
+  ip_allocation_policy {
+    cluster_ipv4_cidr_block  = local.pod_cidr_block == "" ? null : local.pod_cidr_block
+    services_ipv4_cidr_block = local.service_cidr_block == "" ? null : local.service_cidr_block
+  }
+
+  # This change will also enable the metadata server on nodes.
+  # go/gke-cluster-pattern#req4.1.1#req1.1.5 (parts of, vTPM is another section)
+  workload_identity_config {
+    workload_pool = "${var.project}.svc.id.goog"
+  }
+
+  authenticator_groups_config {
+    # Contact safer-gcp to get your group whitelisted for access.
+    # Beta feaure: don't depend on it for breakglass access.
+    security_group = "gke-security-groups@google.com"
+  }
+
+  # GKE Dataplane V2 support
+  # https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#datapath_provider
+  datapath_provider = var.enable_dataplane_v2 ? "ADVANCED_DATAPATH" : "DATAPATH_PROVIDER_UNSPECIFIED"
+
+  # regular release is required for all 1.24+ features.
+  release_channel {
+    channel = "REGULAR"
+  }
+
+  # Allow planned maintenance during this time period.
+  maintenance_policy {
+    recurring_window {
+      start_time = "2023-01-01T09:00:00Z"
+      end_time = "2025-01-01T17:00:00Z"
+      # On the 1st Tuesday of every 6 months.
+      recurrence = "FREQ=MONTHLY;INTERVAL=6;BYDAY=1TU"
+    }
+  }
+
+  addons_config {
+    # We optionally enable the Istio Add-on. Istio is required if there is
+    # any pod-to-pod communication. We keep it optional
+    # to simplify small deployments where there is no pod-to-pod communication
+    # (for now)
+    #
+    # go/gke-cluster-pattern#req5.2.1 for internal traffic, and satified
+    # only if var.enable_istio = "true".
+    istio_config {
+      # Awkward disable flag.
+      disabled = false
+      auth     = "AUTH_MUTUAL_TLS"
+    }
+
+    gce_persistent_disk_csi_driver_config {
+      enabled = true
+    }
+  }
+
+  lifecycle {
+    # Ignore all changes to the default node pool. It's being removed
+    # after creation anyway.
+    ignore_changes = [
+      node_config
+    ]
+  }
+
+  logging_service    = "logging.googleapis.com/kubernetes"
+  monitoring_service = "monitoring.googleapis.com/kubernetes"
+}
+
+# We define explict node pools, so that it can be modified without
+# having to destroy the entire cluster.
+resource "google_container_node_pool" "gke-node-pools" {
+  provider = google-beta
+
+  for_each = {
+    for node_pool in var.node_pools : node_pool.name => node_pool
+  }
+
+  project            = var.project
+  name               = each.value.name
+  cluster            = google_container_cluster.gke-cluster.name
+  location           = var.zone
+  initial_node_count = each.value.nodes_initial
+
+  upgrade_settings {
+    max_surge       = 0
+    max_unavailable = 1
+  }
+
+  management {
+    auto_repair  = false
+    # auto_upgrade needs to be true for release cahnnel "Regular"
+    auto_upgrade = true
+  }
+
+  autoscaling {
```

Comment:
Is this the best way to disable auto-scaling? We don't want it in our case.

Reply:
autoscaling is enabled now. We will have to make min and max same to disable it. But I dont know the impact of that actually.